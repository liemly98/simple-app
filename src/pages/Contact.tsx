import React from "react";
import { useForm } from "react-hook-form";
import toast from "react-hot-toast";

function Contact() {
  const {
    register,
    handleSubmit,
    watch,
    formState: { errors },
  } = useForm();
  const onSubmit = (data: any) => {
    toast.success("Contact created successfully!", {
      position: "top-right",
    });
  };

  return (
    <div className="flex items-center justify-center ">
      <div className="w-full mx-auto px-2 sm:px-6 lg:px-8">
        <div className="w-full">
          <form onSubmit={handleSubmit(onSubmit)}>
            <div className="form-control w-full">
              <label className="label">
                <span className="label-text">Fullname</span>
              </label>
              <input
                type="text"
                {...register("fullname")}
                placeholder="Type here"
                className="input input-bordered w-full"
              />
            </div>

            <div className="form-control w-full">
              <label className="label">
                <span className="label-text">Email</span>
              </label>
              <input
                type="text"
                {...register("email")}
                placeholder="Type here"
                className="input input-bordered w-full"
              />
            </div>

            <div className="form-control">
              <label className="label">
                <span className="label-text">Content</span>
              </label>
              <textarea
                {...register("content")}
                className="textarea textarea-bordered h-24"
                placeholder="Bio"
              ></textarea>
            </div>

            <div className="form-control">
              <label className="label"></label>
              <button type="submit" className="btn btn-secondary btn-outline">
                Submit
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  );
}

export default Contact;
