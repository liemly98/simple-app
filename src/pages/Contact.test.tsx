import React from "react";
import { render, screen, act } from "@testing-library/react";
import Contact from "./Contact";
import { BrowserRouter } from "react-router-dom";
import userEvent from "@testing-library/user-event";

test("renders render contact component", () => {
  const view = render(
    <BrowserRouter>
      <Contact />
    </BrowserRouter>
  );

  // eslint-disable-next-line testing-library/no-unnecessary-act
  act(() => {
    userEvent.click(screen.getByText("Submit"));
  });

  expect(view).toMatchSnapshot();
});
