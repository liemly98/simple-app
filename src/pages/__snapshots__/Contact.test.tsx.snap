// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders render contact component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex items-center justify-center "
      >
        <div
          class="w-full mx-auto px-2 sm:px-6 lg:px-8"
        >
          <div
            class="w-full"
          >
            <form>
              <div
                class="form-control w-full"
              >
                <label
                  class="label"
                >
                  <span
                    class="label-text"
                  >
                    Fullname
                  </span>
                </label>
                <input
                  class="input input-bordered w-full"
                  name="fullname"
                  placeholder="Type here"
                  type="text"
                />
              </div>
              <div
                class="form-control w-full"
              >
                <label
                  class="label"
                >
                  <span
                    class="label-text"
                  >
                    Email
                  </span>
                </label>
                <input
                  class="input input-bordered w-full"
                  name="email"
                  placeholder="Type here"
                  type="text"
                />
              </div>
              <div
                class="form-control"
              >
                <label
                  class="label"
                >
                  <span
                    class="label-text"
                  >
                    Content
                  </span>
                </label>
                <textarea
                  class="textarea textarea-bordered h-24"
                  name="content"
                  placeholder="Bio"
                />
              </div>
              <div
                class="form-control"
              >
                <label
                  class="label"
                />
                <button
                  class="btn btn-secondary btn-outline"
                  type="submit"
                >
                  Submit
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex items-center justify-center "
    >
      <div
        class="w-full mx-auto px-2 sm:px-6 lg:px-8"
      >
        <div
          class="w-full"
        >
          <form>
            <div
              class="form-control w-full"
            >
              <label
                class="label"
              >
                <span
                  class="label-text"
                >
                  Fullname
                </span>
              </label>
              <input
                class="input input-bordered w-full"
                name="fullname"
                placeholder="Type here"
                type="text"
              />
            </div>
            <div
              class="form-control w-full"
            >
              <label
                class="label"
              >
                <span
                  class="label-text"
                >
                  Email
                </span>
              </label>
              <input
                class="input input-bordered w-full"
                name="email"
                placeholder="Type here"
                type="text"
              />
            </div>
            <div
              class="form-control"
            >
              <label
                class="label"
              >
                <span
                  class="label-text"
                >
                  Content
                </span>
              </label>
              <textarea
                class="textarea textarea-bordered h-24"
                name="content"
                placeholder="Bio"
              />
            </div>
            <div
              class="form-control"
            >
              <label
                class="label"
              />
              <button
                class="btn btn-secondary btn-outline"
                type="submit"
              >
                Submit
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
